tasks/set_vars.yml

```diff
@@ -16,3 +16,23 @@
   vars:
     __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
   when: __mssql_vars_file is file
+
+- name: Determine if system is booted with systemd
+  when: not __mssql_is_booted is defined
+  block:
+    - name: Run systemctl
+      # noqa command-instead-of-module
+      command: systemctl is-system-running
+      register: __is_system_running
+      changed_when: false
+      failed_when: false
+
+    - name: Require installed systemd
+      fail:
+        msg: "Error: This role requires systemd to be installed."
+      when: __is_system_running.msg is defined and "No such file or directory" in __is_system_running.msg
+
+    - name: Set flag to indicate that systemd runtime operations are available
+      set_fact:
+        # see https://www.man7.org/linux/man-pages/man1/systemctl.1.html#:~:text=is-system-running%20output
+        __mssql_is_booted: "{{ __is_system_running.stdout not in ['offline', 'degraded'] }}"
```

Comment:
I know that we have used this idiom in other roles - but we have also seen cases e.g. logging role where the system was clearly booted and mostly running, but `systemctl is-system-running` returned `degraded` because journald was not running.  Have you seen cases where `systemctl is-system-running` returns `degraded` during bootc build?

Reply:
No, I haven't. Indeed you are right, "degraded" here is nonsense :see_no_evil: Maybe I meant "unknown"? But anyway, let's only consider "offline" here. Fixed, and I'll send PRs for the three roles which already have that. Thanks for spotting!

Reply:
Sent https://github.com/linux-system-roles/tlog/pull/166 https://github.com/linux-system-roles/logging/pull/449 https://github.com/linux-system-roles/cockpit/pull/217